pkg/kubelet/cm/cpumanager/policy_test.go

```diff
@@ -73,6 +73,32 @@ var (
 		},
 	}
 
+	topoTripleSocketHT = &topology.CPUTopology{
```

Comment:
I think this is a good testcases to have, and I'm all for it; but I think this testcase deserve a comment stating that we are making up a topology to exercise the code rather than modeling existing topologies implemented in hardware.

Reply:
Agreed. I just cherrypicked this and add the numa info, but I should take a closer look at it. Maybe there’s an actual machine out there that would exercise the case as well.

Reply:
I discover the bug while i adding a new comparasion item(cache id) beyond the `socket core number` and  `socket id`, and encouter a strange ordered sort, which incurs my doubt.

IMHO,  the problem could be in sort with more compare dimension and dimensions behaves unsorted(some asend, some disend), like my example.

And yes, the topology example seems strange, no hardware the same, so, what about
1. Add some descriptions for futher clarify? or
2. Add the related testcase while my feature PR pushed with another comparsion which almost 100% brings out the problem? do not need the triple socket toplogy added more, exists topology also meets problem. or
3. Try to find out the hardware encouters the problem, a chanllege to me, i've tried.

thanks and feels pity to bring troubles in commit to you.  

@fromanirh  @klueska

Reply:
> I discover the bug while i adding a new comparasion item(cache id) beyond the `socket core number` and `socket id`, and encouter a strange ordered sort, which incurs my doubt.
> 
> IMHO, the problem could be in sort with more compare dimension and dimensions behaves unsorted(some asend, some disend), like my example.

Interesting, I need to think a bit about these points.
 
> And yes, the topology example seems strange, no hardware the same, so, what about

I'm all for testing boundaries and adding made-up topologies to exercise the code more. So conceptually this is fine by me as long as...

>     1. Add some descriptions for futher clarify? or

...fake topologies used for testing are clearly marked as such. Is completely fine by me to have a bunch of those clearly labeled "for testing purposes only". Simply a comment is more than enough here.

>     2. Add the related testcase while my feature PR pushed with another comparsion which almost 100% brings out the problem? do not need the triple socket toplogy added more, exists topology also meets problem. or

This is another fine option. Having a (variation of) an existing topology to expose the problem is also fine.
I actually expect (and I'd happy to provide when I get the chance) more testing topologies to be added to the tests.

For this specific fix, both options are fine, whatever work best to make this PR progress.

Reply:
@fromanirh , labels added in my own PR https://github.com/kubernetes/kubernetes/pull/101948/files#diff-e5919c496d8843322ad8366c56d5d7f188477902ac3eb161759e785e3b22c0edR60
a cherry-pick update is need from @klueska